File: src/androidTest/java/com/todoroo/andlib/utility/RelativeDayTest.java
package com.todoroo.andlib.utility;

import android.test.AndroidTestCase;

import org.tasks.time.DateTime;

import java.util.Locale;

import static com.todoroo.andlib.utility.DateUtilities.getRelativeDay;
import static org.tasks.Freeze.freezeAt;
import static org.tasks.Freeze.thaw;

public class RelativeDayTest extends AndroidTestCase {

    private static Locale defaultLocale;
    private static final DateTime now = new DateTime(2013, 12, 31, 11, 9, 42, 357);

    @Override
    public void setUp() {
        defaultLocale = Locale.getDefault();
        Locale.setDefault(Locale.US);
        freezeAt(now);
    }

    @Override
    public void tearDown() {
        Locale.setDefault(defaultLocale);
        thaw();
    }

    public void testRelativeDayIsToday() {
        checkRelativeDay(new DateTime(), "Today", "Today");
    }

    public void testRelativeDayIsTomorrow() {
        checkRelativeDay(new DateTime().plusDays(1), "Tomorrow", "Tmrw");
    }

    public void testRelativeDayIsYesterday() {
        checkRelativeDay(new DateTime().minusDays(1), "Yesterday", "Yest");
    }

    public void testRelativeDayTwo() {
        checkRelativeDay(new DateTime().minusDays(2), "Sunday", "Sun");
        checkRelativeDay(new DateTime().plusDays(2), "Thursday", "Thu");
    }

    public void testRelativeDaySix() {
        checkRelativeDay(new DateTime().minusDays(6), "Wednesday", "Wed");
        checkRelativeDay(new DateTime().plusDays(6), "Monday", "Mon");
    }

    public void testRelativeDayOneWeek() {
        checkRelativeDay(new DateTime().minusDays(7), "Dec 24", "Dec 24");
    }

    public void testRelativeDayOneWeekNextYear() {
        checkRelativeDay(new DateTime().plusDays(7), "Jan 7\n2014", "Jan 7\n2014");
    }

    private void checkRelativeDay(DateTime now, String full, String abbreviated) {
        assertEquals(full, getRelativeDay(getContext(), now.getMillis(), false));
        assertEquals(abbreviated, getRelativeDay(getContext(), now.getMillis(), true));
    }
}

File: app/src/androidTest/java/com/todoroo/andlib/utility/RelativeDayTest.kt
package com.todoroo.andlib.utility

import androidx.test.ext.junit.runners.AndroidJUnit4
import kotlinx.coroutines.runBlocking
import org.junit.After
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.tasks.Freeze
import org.tasks.kmp.org.tasks.time.DateStyle
import org.tasks.kmp.org.tasks.time.getRelativeDay
import org.tasks.time.DateTime
import java.util.Locale

@RunWith(AndroidJUnit4::class)
class RelativeDayTest {
    private lateinit var defaultLocale: Locale
    private val now = DateTime(2013, 12, 31, 11, 9, 42, 357)
    
    @Before
    fun setUp() {
        defaultLocale = Locale.getDefault()
        Locale.setDefault(Locale.US)
        Freeze.freezeAt(now)
    }

    @After
    fun tearDown() {
        Locale.setDefault(defaultLocale)
        Freeze.thaw()
    }

    @Test
    fun testRelativeDayIsToday() {
        checkRelativeDay(DateTime(), "Today", "Today")
    }

    @Test
    fun testRelativeDayIsTomorrow() {
        checkRelativeDay(DateTime().plusDays(1), "Tomorrow", "Tmrw")
    }

    @Test
    fun testRelativeDayIsYesterday() {
        checkRelativeDay(DateTime().minusDays(1), "Yesterday", "Yest")
    }

    @Test
    fun testRelativeDayTwo() {
        checkRelativeDay(DateTime().minusDays(2), "Sunday", "Sun")
        checkRelativeDay(DateTime().plusDays(2), "Thursday", "Thu")
    }

    @Test
    fun testRelativeDaySix() {
        checkRelativeDay(DateTime().minusDays(6), "Wednesday", "Wed")
        checkRelativeDay(DateTime().plusDays(6), "Monday", "Mon")
    }

    @Test
    fun testRelativeDayOneWeek() {
        checkRelativeDay(DateTime().minusDays(7), "December 24", "Dec 24")
    }

    @Test
    fun testRelativeDayOneWeekNextYear() {
        checkRelativeDay(DateTime().plusDays(7), "January 7, 2014", "Jan 7, 2014")
    }

    private fun checkRelativeDay(now: DateTime, full: String, abbreviated: String) = runBlocking {
        assertEquals(
                full,
                getRelativeDay(now.millis, DateStyle.LONG))
        assertEquals(
                abbreviated,
                getRelativeDay(now.millis))
    }
}